### ISSUE TITLE
[Bug] HttpClient.Timeout property being ignored when using ProgressMessageHandler

### ISSUE THREAD

**Issue by lobbo232** [xamarin/Xamarin.Forms#7585]
### Description
When using HttpClient with a handler passed in to the constructor for a get request with a timeout set to 10 minutes, the get request will actually timeout after 90 seconds.
If i use a HttpClient without any handlers then the timeout behaves as expected.

### Steps to Reproduce

1. Create a new blank Xamarin Forms project for iOS and Android
2. Create a method triggered by the appearing event in MainPage.xaml.cs
3. In the method create a new ProgressMessageHandler and new HttpClient which takes in the ProgressMessageHandler.
4. Call an API method get request which has a Thread.Sleep in to simulate server processing.
5. Wait for Timeout. It should timeout after 10 minutes but it will timeout after ~90 seconds.

### Expected Behavior
Timeout should be at 10 minute mark.

### Actual Behavior
Timeout is at ~90 seconds mark.

### Basic Information

- Version with issue:
- Last known good version:
- IDE: Visual Studio Community 2019 16.2.5
- Platform Target Frameworks: Xamarin iOS & Android (4.1.0.555618)
  - iOS: 12.4.1
  - Android: 9.0 Pie
- Android Support Library Version: 28.0.0.1
- Nuget Packages: 
- Affected Devices: all iOS and all Android

### Reproduction Link
[HttpClientTimeoutBug.zip](https://github.com/xamarin/Xamarin.Forms/files/3630573/HttpClientTimeoutBug.zip)

**Comment by samhouts** [comment]
@lobbo232 Do you experience the same issue with UWP? Have you tried this in a Xamarin.iOS or Xamarin.Android application without Xamarin.Forms? I suspect this is a mono issue, since we don't have much control over this area in Xamarin.Forms.

**Comment by lobbo232** [comment]
@samhouts I'm sorry but I do not develop for UWP and only use Xamarin.Forms currently. As this is directly related to my job I won't have the time to try other things out for this. Attached to the original post is a solution with the minimum code to recreate the issue.

**Comment by PureWeen** [comment]
@lobbo232 I'm wondering if the end point you're testing against has a time out?

I tested your project and I was able to get the request to wait longer than 90 seconds. 

The end point I was testing against would kill the web thread after 4 minutes and when I used your sample to point at that (just a basic azure website) it would sit for 4 minutes until the server killed the thread.

I'm also running VS 16.3 and there were some recent mono changes so perhaps try the latest 16.3 which is running Android 10. Try updating the project to Android 10? Make sure you have the http handler for the Android project configured for Native?

Let me know if any of the above suggestions work for you
